src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
+            resultArray[x] = resultArray[x] * 2;
+            gameScore.textContent = +gameScore.textContent + resultArray[x];
+
+            resultArray.splice((x - 1), 1);
+            x = 0;
+          }
+        };
+
+        let indexField = 3;
+
+        for (let index = resultArray.length - 1; index >= 0; index--) {
+          currentRow.children[indexField].textContent = resultArray[index];
+          indexField = indexField - 1;
+        }
+
+        resultArray.splice();
+      }
+    }
+
+    function ArrowUp() {
+      for (let i = 0; i < 4; i++) {
+        const resultArray = [];
+
+        for (let y = 0; y < 4; y++) {
+          const currentChild = fieldRowList[y].children[i];
+
+          if (currentChild.textContent.length !== 0) {
+            resultArray.push(+currentChild.textContent);
+            currentChild.textContent = '';
+          }
+        };
+
+        for (let x = resultArray.length - 1; x >= 0; x--) {
+          if (resultArray[x - 1] === undefined) {
+            break;
+          }
+
+          if (resultArray[x] === resultArray[x - 1]) {
+            resultArray[x] = resultArray[x] * 2;
+            gameScore.textContent = +gameScore.textContent + resultArray[x];
+
+            resultArray.splice((x - 1), 1);
+            x = 0;
+          }
+        };
+
+        for (let index = 0; index < resultArray.length; index++) {
+          fieldRowList[index].children[i].textContent = resultArray[index];
+        }
+
+        resultArray.splice();
+      }
+    }
+
+    function ArrowDown() {
+      for (let i = 0; i < 4; i++) {
+        const resultArray = [];
+
+        for (let y = 0; y < 4; y++) {
+          const currentChild = fieldRowList[y].children[i];
+
+          if (currentChild.textContent.length !== 0) {
+            resultArray.push(+currentChild.textContent);
+            currentChild.textContent = '';
+          }
+        };
+
+        for (let x = resultArray.length - 1; x >= 0; x--) {
+          if (resultArray[x - 1] === undefined) {
+            break;
+          }
+
+          if (resultArray[x] === resultArray[x - 1]) {
+            resultArray[x] = resultArray[x] * 2;
+            gameScore.textContent = +gameScore.textContent + resultArray[x];
+
+            resultArray.splice((x - 1), 1);
+            x = 0;
+          }
+        };
+
+        let indexField = resultArray.length - 1;
+
+        for (let index = 3; index > 0; index--) {
+          fieldRowList[index].children[i].textContent = resultArray[indexField];
+          indexField = indexField - 1;
+        }
+
+        resultArray.splice();
+      }
+    }
+
+    function ColorField() {
+      for (const fieldCell of fieldCellList) {
+        const textCon = fieldCell.textContent;
+
+        if (fieldCell.textContent > 0) {
+          fieldCell.className = `field-cell field-cell--${textCon}`;
+        } else {
+          fieldCell.className = 'field-cell';
+        }
+      }
+    }
+
+    ColorField();
+
+    setTimeout(() => {
```

Comment:
why do we need here timeout?

Reply:
so that the human eye can see a smooth addition